Geometry/VeryForwardGeometryBuilder/src/CTPPSGeometry.cc

```diff
@@ -0,0 +1,207 @@
+/****************************************************************************
+*
+* This is a part of TOTEM offline software.
+* Authors: 
+*  Jan Kaspar (jan.kaspar@gmail.com) 
+*
+****************************************************************************/
+
+#include "Geometry/VeryForwardGeometryBuilder/interface/CTPPSGeometry.h"
+
+//----------------------------------------------------------------------------------------------------
+
+char
+CTPPSGeometry::build( const DetGeomDesc* gD )
+{
+  // reset
+  theMap.clear();
+  theRomanPotMap.clear();
+  stationsInArm.clear();
+  rpsInStation.clear();
+  detsInRP.clear();
+
+  // propagate through the GeometricalDet structure and add all detectors to 'theMap'
+  std::deque<const DetGeomDesc *> buffer;
+  buffer.emplace_back(gD);
+  while ( !buffer.empty() ) {
+    const DetGeomDesc *d = buffer.front();
+    buffer.pop_front();
+
+    // check if it is a sensor
+    if ( d->name().name() == DDD_TOTEM_RP_SENSOR_NAME
+      || d->name().name() == DDD_CTPPS_DIAMONDS_SEGMENT_NAME
+      || d->name().name() == DDD_CTPPS_PIXELS_SENSOR_NAME ) {
+	  addSensor(d->geographicalID(), d);
+    }
+
+    // check if it is a RP
+    if ( d->name().name() == DDD_TOTEM_RP_RP_NAME
+      || d->name().name() == DDD_CTPPS_PIXELS_RP_NAME
+      || d->name().name() == DDD_CTPPS_DIAMONDS_RP_NAME ) {
+      addRP( d->geographicalID(), d );
+    }
+    
+    for ( unsigned int i = 0; i < d->components().size(); i++ )
+      buffer.emplace_back( d->components()[i] );
+  }
+
+  // build sets
+  for ( const auto& it : theMap ) {
+    const CTPPSDetId detId( it.first );
+    const CTPPSDetId rpId = detId.getRPId();
+    const CTPPSDetId stId = detId.getStationId();
+    const CTPPSDetId armId = detId.getArmId();
+
+    stationsInArm[armId].insert( armId );
+    rpsInStation[stId].insert( rpId );
+    detsInRP[rpId].insert( detId );
+  }
+
+  return 0;
+}
+
+//----------------------------------------------------------------------------------------------------
+
+char
```

Comment:
Changed it to `bool` and reverted the return type (`true` = successfully added ; `false` = already present)